app/controllers/pages_controller.rb

```diff
@@ -5,14 +5,13 @@ def home
   end
 
   def profile
-    @user = current_user
-    @user_bookings = Booking.where(user: @user)
+    @user_bookings = Booking.where(user: current_user)
+    @pets = Pet.where(pet_owner: current_user)
   end
 
   def change_role
-    @user = current_user
-    @user.role = params[:user][:role]
-    @user.save
+    current_user.role = params[:user][:role]
```

Comment:
Can we discuss it together tomorrow morning? I'd like more explanations, so if you can guide us through the changes that'd be great!